### ISSUE TITLE
Execution error when running (auth!)

### ISSUE THREAD

**Issue by jtlocsei** [timothypratley/happygapi#3]
Hi Timothy, 

I love the idea of this library! One library to rule them all instead of struggling with the Java client library for each specific Google API. Amazing! I'm keen to get it to work, but I'm having trouble getting `(auth!)` to work. 

Steps to reproduce: 
1. Create a Google service account and save the key as `service.json` in the project directory. I know this file is readable because I can run this code snippet from the `init!` function in `oauth2_credentials.clj`.
```
(let [service-file (io/file "service.json")]
              (when (.exists service-file)
                (json/parse-string (slurp service-file) true)))
```
and the result is a map with these keys 
```
(:client_x509_cert_url
 :token_uri
 :type
 :private_key_id
 :auth_provider_x509_cert_url
 :client_email
 :client_id
 :project_id
 :private_key
 :auth_uri)
```

2.  In my REPL, run 
```
(credentials/init! ["https://www.googleapis.com/auth/analytics.readonly"])
```
The result is:
```
=> #object[happy.oauth2_credentials$save_credentials 0x626fcfd8 "happy.oauth2_credentials$save_credentials@626fcfd8"]
```

3. In my REPL, run `(credentials/auth!)`. The result is:
```
Execution error (NullPointerException) at java.util.regex.Matcher/getTextLength (Matcher.java:1770).
null
```
The stack trace is:
```
java.lang.NullPointerException: null
 at java.util.regex.Matcher.getTextLength (Matcher.java:1770)
    java.util.regex.Matcher.reset (Matcher.java:416)
    java.util.regex.Matcher.<init> (Matcher.java:253)
    java.util.regex.Pattern.matcher (Pattern.java:1133)
    java.util.regex.Pattern.split (Pattern.java:1261)
    java.util.regex.Pattern.split (Pattern.java:1334)
    clojure.string$split.invokeStatic (string.clj:224)
    clojure.string$split.invoke (string.clj:219)
    happy.oauth2$credential_scopes.invokeStatic (oauth2.clj:134)
    happy.oauth2$credential_scopes.invoke (oauth2.clj:133)
    happy.oauth2_capture_redirect$update_credentials.invokeStatic (oauth2_capture_redirect.clj:47)
    happy.oauth2_capture_redirect$update_credentials.invoke (oauth2_capture_redirect.clj:45)
    happy.oauth2_credentials$auth_BANG_.invokeStatic (oauth2_credentials.clj:41)
    happy.oauth2_credentials$auth_BANG_.invoke (oauth2_credentials.clj:37)
    happy.oauth2_credentials$auth_BANG_.invokeStatic (oauth2_credentials.clj:38)
    happy.oauth2_credentials$auth_BANG_.invoke (oauth2_credentials.clj:37)
```

It looks to me like the problem is that the `auth!` function sets `credentials` to `(@*fetch user)` which is `nil` because at this point the user credentials have not yet been saved to disk into an edn file. Then `(ocr/update-credentials @*secret credentials @*scopes nil)` passes a nil as the `credentials` argument which then gets passed through to `happy.oauth2/credential-scopes` where `str/split` chokes on the `nil` argument. 

Am I using this library wrong somehow? What do I do to get it to work? Thanks in advance for any guidance!

**Comment by mkrcah** [comment]
I'm running into the same `NullPointerException`, when parsing a file myself:
```clojure
(def required-scopes
  ["https://www.googleapis.com/auth/spreadsheets"
   "https://www.googleapis.com/auth/drive"])

(def credentials
    (read-json! "...path to service.json..."))
       
(credentials/init! credentials required-scopes)
(credentials/auth!)
```


UPDATE: Got the auth token by calling the `refresh-credentials` directly. Here's a quick fiddle as I was exploring the api:

```clojure
; getting oauth token
(def credentials (read-json! credentials-f))

(def required-scopes 
  ["https://www.googleapis.com/auth/spreadsheets"])

(defn get-access-token! [credentials required-scopes]
  (happy.oauth2/refresh-credentials credentials required-scopes {}))

; requests
(defn auth-header [{:keys [access_token]}]
  {:headers {"Authorization" (str "Bearer " access_token)}})

(defn get-spreadsheet-req [id]
  {:spreadsheetId id})

; here we go
(def access-token (get-access-token! credentials required-scopes))
(gsheets/get$ (auth-header access-token)
                (get-spreadsheet-req spreadsheet-id)))
```

**Comment by timothypratley** [comment]
Thanks to @nickdex for fixing this https://github.com/timothypratley/happygapi/pull/4
released as 0.4.9
